pubsub/google/cloud/pubsub/_gax.py

```diff
@@ -413,8 +413,15 @@ def subscription_pull(self, subscription_path, return_immediately=False,
                 subscription_path, max_messages,
                 return_immediately=return_immediately)
         except GaxError as exc:
-            if exc_to_code(exc.cause) == StatusCode.NOT_FOUND:
+            code = exc_to_code(exc.cause)
+            if code == StatusCode.NOT_FOUND:
                 raise NotFound(subscription_path)
+            elif exc_to_code(exc.cause) == StatusCode.DEADLINE_EXCEEDED:
```

Comment:
@daspecster You have repeated the "computation". That's why `code` is created before you start checking